Login Refactoring

pondorasti opened the pull request:
* Front Page Refactoring
* Login Refactoring
* Sign-up Refactoring
* UIInputView class
* Notification Extension

Commit:
WIP: UIInputView class

+ KFCLogin Refactoring

Commit:
Done: Login Refactoring

+ Notification Extension

Commit:
Done: Refactoring FontPage

Commit:
Done: RegisterForm Refactoring

Commit:
Minor Update

LinnierGames commented on KifuSF/Notification.swift:
```diff
@@ -0,0 +1,16 @@
+
+extension Notification {
+    func getKeyboardHeight() -> CGFloat? {
+        return (self.userInfo?[UIKeyboardFrameEndUserInfoKey] as? NSValue)?.cgRectValue.height
```
nice

LinnierGames commented on KifuSF/RegisterFormViewController.swift:
```diff
@@ -98,16 +106,15 @@ class KFCRegisterForm: UIViewController {
+            let username = usernameInputView.textFieldContainer.textField.text, !username.isEmpty,
+            let contactNumber = phoneNumberInputView.textFieldContainer.textField.text, !contactNumber.isEmpty,
+            let email = emailInputView.textFieldContainer.textField.text, !email.isEmpty,
+            let password = passwordInputView.textFieldContainer.textField.text, !password.isEmpty else {
```
What do you think of this as an exension?

```swift
extension String {
    
    var isNotEmpty: Bool {
        return !self.isEmpty
    }
    
    /**
     checks if self is empty. If so, return the given default string.
     
     - parameter defaultString: return this if self is empty or nil
     
     - returns: either return self if not an empty string or return the given default value
     */
    func ifEmpty(use defaultString: String) -> String {
        if self.isEmpty {
            return defaultString
        } else {
            return self
        }
    }
}

extension Optional where Wrapped == String {
    
    /**
     unwraps and checks if the string is empty. If so, return the given default string.
     
     This is similar to using the nil-colicent but also checks `.isEmpty`
     
     - parameter defaultString: return this if self is empty or nil
     
     - returns: either return self if not nil and not an empty string or return the given default value
     */
    func ifEmptyOrNil(use defaultString: String) -> String {
        if let unwrappedSelf = self, unwrappedSelf.isEmpty == false {
            return unwrappedSelf
        } else {
            return defaultString
        }
    }
}
```

LinnierGames reviewed (approved):
Great stuff! I've added some code for syntax-tic sugar

LinnierGames commented:
@Pondorasti The safe area is not used in the cover screen

LinnierGames commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Login Flow Refactoring](https://trello.com/c/SdNFulY3/114-login-flow-refactoring)

Commit:
Minor Refactoring

Change: UIConfigurable method name

Learned about directionalLayoutMargins and integrated them
